Mathlib/CategoryTheory/Sites/EqualizerSheafCondition.lean

```diff
@@ -259,6 +261,100 @@ theorem sheaf_condition : R.IsSheafFor P ↔ Nonempty (IsLimit (Fork.ofι _ (w P
     simp [forkMap]
 #align category_theory.equalizer.presieve.sheaf_condition CategoryTheory.Equalizer.Presieve.sheaf_condition
 
+namespace Arrows
+
+open Presieve
+
+variable {B : C} {I : Type} (X : I → C) (π : (i : I) → X i ⟶ B) [UnivLE.{w, max v u}]
+    [(ofArrows X π).hasPullbacks]
+-- TODO: allow `I : Type w` 
+
+/--
+The middle object of the fork diagram of <https://stacks.math.columbia.edu/tag/00VM>.
+-/
+def FirstObj : Type max v u := ∏ (fun i ↦ P.obj (op (X i)))
```

Comment:
Here we're [perpetuating the equalizer condition](https://github.com/leanprover-community/mathlib4/pull/7654#issuecomment-1767913320) by introducing more API. It's not quite objectionable here because Type* has products so there's no extra assumption; but could it be done (or done better) using the Pairwise diagram (whose API is lacking)? @erdOne surely knows more than me.

Reply:
Yeah. I wonder if it is possible to mimic [`CategoryTheory.Presheaf.isSheaf_iff_multifork`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Sites/Sheaf.html#CategoryTheory.Presheaf.isSheaf_iff_multifork) instead.

Reply:
@alreadydone said "The only exception is probably when all intersections are empty, so the limit is actually a product" in the linked PR. This is precisely my application, see #7804 and #7949.

I'm not against using an alternative approach (I haven't looked into it), I'm just pointing out that maybe your objections don't apply to my application.

Reply:
What is the difference of this `FirstObj` with the one above in the file? It is that it is for a `Presieve.ofArrows` instead that for a general presieve?

Reply:
The difference is that presieves don't see duplicates, whereas families of arrows do. So for example if I have a family consisting of the same arrow twice, the product in the `Presieve` version of `FirstObj` has a singleton indexing set, while the product in this `FirstObj` will be of two copies of the same thing.

Reply:
> The only exception is probably when all intersections are empty, so the limit is actually a product" in the linked PR. This is precisely my application, see https://github.com/leanprover-community/mathlib4/pull/7804 and https://github.com/leanprover-community/mathlib4/pull/7949.

Okay, I won't argue against the approach then :)